lib/ci-helper.ts

```diff
@@ -626,15 +638,43 @@ export class CIHelper {
         }
     }
 
+    public async checkCommits(pr: IPullRequestInfo):
+        Promise<string> {
+        let comment: string = "";
+        const maxCommits = 30;
+        if (pr.commits && pr.commits > maxCommits) {
+            comment = `The pull request has ${pr.commits
+                } commits.  The max allowed is ${maxCommits
+                }.  Please squash some commits and 'git push --force'`;
+        }
+
+        return comment;
+    }
+
     public async handlePush(repositoryOwner: string, prNumber: number) {
         const pr = await this.github.getPRInfo(repositoryOwner, prNumber);
-        const gitGitGadget = await GitGitGadget.get(".");
-        if (!pr.hasComments && !gitGitGadget.isUserAllowed(pr.author)) {
-            const pullRequestURL =
-                `https://github.com/${repositoryOwner}/git/pull/${prNumber}`;
-            const welcome = (await readFile("res/WELCOME.md")).toString()
-                .replace(/\${username}/g, pr.author);
-            this.github.addPRComment(pullRequestURL, welcome);
+        const pullRequestURL = `https://github.com/${repositoryOwner
+                                }/git/pull/${prNumber}`;
+
+        try {
+            const gitGitGadget = await GitGitGadget.get(this.gggConfigDir,
+                                                        this.workDir);
```

Comment:
Sneaky ;-) It's a good change to replace `"."` here, of course.

Reply:
Missed it last time around.